onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2241,6 +2249,22 @@ TensorrtExecutionProvider::GetCapability(const GraphViewer& graph,
   // Generate unique kernel name for TRT graph
   HashValue model_hash = TRTGenerateId(graph);
 
+  SubGraph_t supported_node_vector = {{}, true};
+  for (int index = 0 ; index < graph.NumberOfNodes() ; ++index) {
+    supported_node_vector.first.push_back(index);
+  }
+  std::unique_ptr<IndexedSubGraph> sub_graph = GetSubGraph(supported_node_vector, graph, model_hash,
```

Comment:
Move this block into `if (weightless_engine_enable_ || single_cached_engine_present)` below?

Reply:
Hmm, but I need to have access to `sub_graph->GetMetaDef()->name()` during the `if` evaluation?

Reply:
okay, true, then let's keep it for now.